lib/web_ui/lib/src/engine/text/canvas_paragraph.dart

```diff
@@ -185,6 +187,66 @@ class CanvasParagraph implements ui.Paragraph {
     return rootElement;
   }
 
+
+  SVGGElement? _cachedSvgElement;
+
+  /// Returns a SVG element that represents the entire paragraph and its
+  /// children.
+  ///
+  /// Generates a new SVG element on every invocation.
+  SVGGElement toSvgElement() {
+    assert(isLaidOut);
+    final SVGGElement? svgElement = _cachedSvgElement;
+    if (svgElement == null) {
+      return _cachedSvgElement ??= _createSvgElement();
+    }
+    return svgElement.cloneNode(true) as SVGGElement;
+  }
+
+  SVGGElement _createSvgElement() {
+    final SVGGElement rootElement = createSVGGElement();
+
+    // Prevent the browser from doing any line breaks in the paragraph. We want
+    // to have full control of the paragraph layout.
+    rootElement.style.whiteSpace = 'pre';
+
+    // Append all spans to the paragraph.
+    for (int i = 0; i < lines.length; i++) {
+      final ParagraphLine line = lines[i];
+      for (final LayoutFragment fragment in line.fragments) {
+        if (fragment.isPlaceholder) {
+          continue;
+        }
+
+        final String text = fragment.getText(this);
+        if (text.trim().isEmpty) {
+          continue;
+        }
+
+        final SVGTextElement spanElement = createSVGTextElement();
+        applyTextStyleToSvgElement(element: spanElement, style: fragment.style);
+
+        final ui.Rect boxRect = fragment.toPaintingTextBox().toRect();
+        spanElement.addX(boxRect.left);
+        // SVG places <text> elements such that the text baseline coincides with
+        // the Y coordinate. However, Flutter positions the top edge of the text
+        // line at the Y coordinate. For the SVG text to match what Flutter
+        // expects, the baseline is added to the Y coordinate.
+        spanElement.addY(boxRect.top + line.baseline);
```

Comment:
(I think you already did this but haven't pushed it yet)
```suggestion
        spanElement.addY(line.baseline);
```